### PR TITLE
- Impossible Traveller playbook, type, fields, layout, scripts, test

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19787
fixes: https://github.com/demisto/etc/issues/20057

## Description
- Added the Impossible Traveller playbook
- Added a test playbook
- Added incident type
- Added incident fields
- Added incident layout
- Added 2 scripts: CalculateTimeDifference and CalculateGeoDistance
- Updated `CompleteManualTask` script to allow specifying closing option of conditional tasks.

## Screenshots
Impossible_Traveller_dev_Sun_Nov_10_2019

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] [Documentation](https://github.com/demisto/etc/issues/20013)
- [ ] Code Review

## Technical writer review
It will be a waste of time to write each line of each file.
The doc review should cover:
- Playbook description
- Task names
- Task descriptions
- Incident field names
- Incident field descriptions
- Script input / output descriptions
- All changelogs

### CODE REVIEW

File: Scripts/CalculateGeoDistance/CalculateGeoDistance.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import geopy.distance
+
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateGeoDistance/CalculateGeoDistance.py]: - [x] Fixed

File: Scripts/CalculateGeoDistance/CalculateGeoDistance.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    entries.append({
+        "Type": entryTypes['note'],
+        "Contents": geo_distance,
+        "ContentsFormat": formats['text'],
+        "HumanReadable": "Calculated Distance: {} miles.".format(str(geo_distance)),
+        "EntryContext": {
+            "Geo.Distance": geo_distance,
+            "Geo.Coordinates": [src_coords, dest_coords]
+        }
+    })
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateGeoDistance/CalculateGeoDistance.py]: - [x] Fixed

File: Scripts/CalculateGeoDistance/CalculateGeoDistance.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+except Exception as ex:
+    entries.append({
+        "Type": entryTypes["error"],
+        "ContentsFormat": formats["text"],
+        "Contents": "Error occurred while parsing output from command. Exception info:\n" + str(ex)
+    })
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateGeoDistance/CalculateGeoDistance.py]: - [x] Fixed

File: Scripts/CalculateGeoDistance/CalculateGeoDistance.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+from CommonServerPython import *
+from CommonServerUserPython import *
+import geopy.distance
+
+entries = []
+
+try:
+    # Extract each set of coordinates and create a list
+    src_coords_list = demisto.args()['src_coords'].split(",")
+    dest_coords_list = demisto.args()['dest_coords'].split(",")
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateGeoDistance/CalculateGeoDistance.py]: - [x] Fixed

File: Scripts/CalculateTimeDifference/CalculateTimeDifference.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        "Type": entryTypes['note'],
+        "Contents": mins,
+        "ContentsFormat": formats['text'],
+        "HumanReadable": "Calculated Time Difference: {} minutes.".format(str(mins)),
+        "EntryContext": {
+            "Time.Difference": mins,
+            "Time.Start": start_time,
+            "Time.End": end_time
+        }
+    })
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateTimeDifference/CalculateTimeDifference.py]: - [x] Fixed

File: Scripts/CalculateTimeDifference/CalculateTimeDifference.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+except Exception as ex:
+    entries.append({
+        "Type": entryTypes["error"],
+        "ContentsFormat": formats["text"],
+        "Contents": "Error occurred while parsing output from command. Exception info:\n" + str(ex)
+    })
+demisto.results(entries)
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateTimeDifference/CalculateTimeDifference.py]: - [x] Fixed

File: Scripts/CalculateGeoDistance/CalculateGeoDistance.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    })
+
+except Exception as ex:
+    entries.append({
+        "Type": entryTypes["error"],
+        "ContentsFormat": formats["text"],
+        "Contents": "Error occurred while parsing output from command. Exception info:\n" + str(ex)
+    })
+
+demisto.results(entries)
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Scripts/CalculateGeoDistance/CalculateGeoDistance.py]: - [x] Fixed

File: Tests/conf.json [hunk 8]
```diff
@@ -1503,6 +1503,16 @@
         {
             "playbookID": "FilterByList - Test",
             "fromversion": "4.5.0"
+        },
+            {
+            "playbookID": "Impossible Traveller - Test",
+            "integrations": [
+            "Ipstack",
+            "Rasterize",
+            "Active Directory Query v2"
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Tests/conf.json]: - [x] Fixed

**Comment by ShahafBenYakir** [comment]
@idovandijk Please pull and check that everything still works after my fixes

**Comment by yaron-libman** [comment]
Updated texts.

**Comment by idovandijk** [comment]
> All incident fields/incident type must be `"fromVersion": "5.0.0"`

Why? We need to use them on 4.5

**Pull request merged**